src/System.Net.Http/tests/FunctionalTests/TestHelper.cs

```diff
@@ -155,5 +156,33 @@ public static void EnsureHttp2Feature(HttpClientHandler handler)
                 BindingFlags.NonPublic | BindingFlags.Instance);
             field_maxHttpVersion.SetValue(_settings, new Version(2, 0));
         }
+
+        public static bool NativeHandlerSupportsSslConfiguration()
+        {
+#if TargetsWindows
+            return true;
+#else
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return false;
+            }
+
+            // For other Unix-based systems it's true if (and only if) the currect openssl backend
+            // is used with libcurl.
+            bool hasAnyOpenSsl =
+                Interop.Http.GetSslVersionDescription()?.StartsWith(Interop.Http.OpenSslDescriptionPrefix, StringComparison.OrdinalIgnoreCase) ?? false;
+
+            if (!hasAnyOpenSsl)
+            {
+                return false;
+            }
+
+            // We're on an OpenSSL-based system, with an OpenSSL backend.
+            // Ask the product how it feels about this.
+            Type interopHttp = typeof(HttpClient).Assembly.GetType("Interop+Http");
+            PropertyInfo hasMatchingOpenSslVersion = interopHttp.GetProperty("HasMatchingOpenSslVersion", BindingFlags.Static | BindingFlags.NonPublic);
+            return (bool)hasMatchingOpenSslVersion.GetValue(null);
```

Comment:
If the tests are pulling in the file anyway, do we need to exclude that code when built in the tests, or could we just use that rather than using reflection?

Reply:
The tests pull in that one file, but not the rest of the work for the HttpNative/CryptoNative initializer sequencing, and leaving it that way helps to avoid a Heisenbug where the tests end up doing the initialization on behalf of the product and the product eventually neglects to do so.